### PR TITLE
Student participation

### PR DESCRIPTION
**Pull request by exacs**
- Get user from Ladok UID
- Continue handling registreringEvent
- Add new Ladok endpoints
- Functions to get terms
- Add recursive types
- Implement "findStudiestruktur"
- Unit tests for utils
- Delete a typ
- Scripts test and test:unit
- Expose type TStudiestruktur
- Change definition for ProgramParticipation
- Update snapshot
- Add tests

### CODE REVIEW

File: packages/ladok-integration/__tests__/example.integration.ts
Comment (jhsware) [inline packages/ladok-integration/__tests__/example.integration.ts:1]: Perhaps we should call this `index.integration.ts` to show that it contains tests for index.ts?

File: packages/ladok-integration/__tests__/utils.test.ts
Comment (jhsware) [inline packages/ladok-integration/__tests__/utils.test.ts:1]: Currently unit tests are called xxx.spec.ts. This is the invoking code from /package.json:

```
"test:regression": "jest --testMatch '**/__tests__/**/*.spec.ts' --testTimeout 1000",
```

File: packages/ladok-integration/package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
   "source": "src/index.ts",
   "scripts": {
     "build": "tsc src/*.ts --outDir dist --sourceMap --declaration",
-    "test": "jest"
+    "test:unit": "jest __tests__/*.test.ts",
```
Comment (jhsware) [inline packages/ladok-integration/package.json:10]: Affected by name change.

File: packages/ladok-integration/script.ts
Comment (jhsware) [inline packages/ladok-integration/script.ts:1]: Remove or add a comment at beginning of file to explain the intended use.

**Review by jhsware — CHANGES_REQUESTED** [review]
Looks good in general! Some minor stuff...

File: packages/ladok-integration/script.ts
Comment (exacs) [reply packages/ladok-integration/script.ts:1]: This was not intended to be pushed :)

**Comment by jhsware** [comment]
LGTM

**Pull request merged**